### ISSUE TITLE
Allow custom color or transparent background on embeds

### ISSUE THREAD

**Issue by cleydsonjr** [metabase/metabase#9856]
It would be nice to be able to set a background color (or transparent) for embedded dashboards.

There is an inline style `background-color: transparent;` on the infame body, but the dashboard is wrapped on a div with this class:
```css
.EmbedFrame {
    background-color: white;
}
```

I'm using the non-standard attribute `allowtransparency` but, because of that div it has no effect. 

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by NilsKlimmChemondis** [comment]
This element also has the following style applied, why it does not cover the whole page but only the size of the iframe window:

```
.spread, .Y2_Z- {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
}
```

**Comment by mlazowik** [comment]
Yeah, one option to make the `background-color` cover the whole page is to push it to the element below. Initially I wanted to report that as a separate issue, but maybe it's not big enough?

<img width="757" alt="Screenshot 2021-01-24 at 17 21 11" src="https://user-images.githubusercontent.com/1978721/105636400-8f606100-5e68-11eb-81cb-b8b3cdf931e8.png">

**Comment by nkeat12** [comment]
is there any workaround or hacks for this? The documentation references theme=transparent as a url param but there is still a white bg within the iframe..

**Comment by ClementNumericite** [comment]
Still no way to just make the main background transparent but the background of viz cards white ? 
It'll be the smoothest theme for integration.

**Comment by izonhen** [comment]
There is a way to custom css design for embedded dashboard?
Using the pro plan

**Comment by paoliniluis** [comment]
@izonhen please write your questions to the support email

**Comment by izonhen** [comment]
What is the email please?

**Comment by gusaiani** [comment]
PR below lists visual design decisions to make before implementing this.

https://github.com/metabase/metabase/pull/27586

**Comment by albertoperdomo** [comment]
I think this is implemented already. @gusaiani ?
I used `#theme=transparent` yesterday to embed a dashboard with transparent background (signed embedding).

![Image](https://github.com/metabase/metabase/assets/24216/2a8cdb90-cb46-4217-bb82-092cb4b92d0d)

**Comment by gusaiani** [comment]
> I think this is implemented already.

Hi @albertoperdomo, I don't know that it's properly implemented.
Perhaps, but it would be a surprise.

Things that come to mind are filter dropdowns and other artifacts like that.

**Comment by albertoperdomo** [comment]
Correct. Thanks for the pointer @gusaiani!

The filter itself is rendered transparent, but the modals are not.

Screenshots

![Image](https://github.com/metabase/metabase/assets/24216/11810b67-f7bd-4a55-9d26-39e3b2dc7725)
![Image](https://github.com/metabase/metabase/assets/24216/09bf3061-914e-4821-a7c5-5605efc5eea0)

**Comment by albertoperdomo** [comment]
Well, now thinking, the modals can't be transparent.... Otherwise they would not be readable....

**Comment by gusaiani** [comment]
Right. I suppose I expressed myself unclearly: are we sure everything that should be readable is readable in a transparent embed?

**Comment by luxemate** [comment]
First I saw the result of choosing Transparent option in the Appearance I thought it's some kind of bug. 
I would suggest making background transparency as a separate checkbox, not on the same level with light and dark. Or maybe as a color picker with ability to set color transparency.
For example, if we have a light theme with a gray background in our CMS it's okay for us to have white panels with a transparent background, but totally transparent everything doesn't make any sense.

**Comment by maxcaron** [comment]
Any updates on this? I need the background of the cards to be white and the outside to be transparent. That is how it is displayed in Metabase. If I set theme=transparent, then everything becomes transparent. If I leave it as is, then the cards are fine but the rest of the iframe has a white background also. 

Two css classes seem to be responsible for this, EmbedFrame and EmbedFrame-header. I can't inject css into the iFrame since it is blocked by CORS making it pretty much impossible to replicate how it is displayed in Metabase vs my iframe.

Is this something that is planned on being fixed or is there a known workaround to this issue?

**Comment by albertoperdomo** [comment]
In upcoming version 51, we deprecate the transparent theme. Instead we introduce the option to choose between the light and dark themes and optionally toggle background transparency.

If you want to have a sneak peak, here's the download link for 51 Beta: https://downloads.metabase.com/enterprise/beta/metabase.jar

I hope this addresses everyone's issues. If not, please let us know!
